Description: Fix build with make 4.3 while retaining compatibility with make 4.2
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Author: Peter Michael Green <plugwash@debian.org>
Last-Update: 2020-07-23

Index: gst-plugins-bad1.0-1.16.2/common/glib-gen.mak
===================================================================
--- gst-plugins-bad1.0-1.16.2.orig/common/glib-gen.mak
+++ gst-plugins-bad1.0-1.16.2/common/glib-gen.mak
@@ -5,7 +5,12 @@
 #glib_enum_define=GST_COLOR_BALANCE
 #glib_enum_prefix=gst_color_balance
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+# GNU make 4.3 changes the escaping rules for the hash sign, the reccomended
+# way to maintain compatibility with both 4.2 and 4.3 is to use a variable
+# see https://lwn.net/Articles/810071/
+HASHINCLUDE:=\#include
+
+enum_headers=$(foreach h,$(glib_enum_headers),\n$(HASHINCLUDE) \"$(h)\")
 
 # these are all the rules generating the relevant files
 %-marshal.h: %-marshal.list
Index: gst-plugins-bad1.0-1.16.2/common/gst-glib-gen.mak
===================================================================
--- gst-plugins-bad1.0-1.16.2.orig/common/gst-glib-gen.mak
+++ gst-plugins-bad1.0-1.16.2/common/gst-glib-gen.mak
@@ -8,7 +8,12 @@
 #glib_gen_decl_banner=GST_EXPORT
 #glib_gen_decl_include=\#include <gst/foo/foo-prelude.h>
 
-enum_headers=$(foreach h,$(glib_enum_headers),\n\#include \"$(h)\")
+# GNU make 4.3 changes the escaping rules for the hash sign, the reccomended
+# way to maintain compatibility with both 4.2 and 4.3 is to use a variable
+# see https://lwn.net/Articles/810071/
+HASHINCLUDE:=\#include
+
+enum_headers=$(foreach h,$(glib_enum_headers),\n$(HASHINCLUDE) \"$(h)\")
 
 # these are all the rules generating the relevant files
 $(glib_gen_basename)-marshal.h: $(glib_gen_basename)-marshal.list
